pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -511,9 +524,16 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject) in
 			subAssignableAdapters.ProcessChange(change)
 
 		case change := <-subAppNetworkConfig.MsgChan():
-			// If we have an NetworkInstanceConfig process it first
-			checkAndProcessNetworkInstanceConfig(&zedrouterCtx)
+			// in modify path
+			// If we have NetworkInstanceConfig event, process it first
+			pending := checkAndProcessNetworkInstanceConfig(change,
+				&zedrouterCtx)
 			subAppNetworkConfig.ProcessChange(change)
+			if pending != nil {
+				for _, change := range pending {
+					subNetworkInstanceConfig.ProcessChange(change)
+				}
+			}
```

Comment:
You shouldn't need any of these changes with the new code to defer the delete of the network instance status.

Reply:
Thanks for removing this